### ISSUE TITLE
test forknotify.py: JSONRPC error: CreateNewBlock(): TestBlockValidity failed

### ISSUE THREAD

**Issue by error10** [bitcoinxt/bitcoinxt#104]
Here we go again... 

```
Running 2nd level testscript forknotify.py...
Initializing test directory /tmp/testc3MPBy
JSONRPC error: CreateNewBlock(): TestBlockValidity failed
  File "/home/error/bitcoinxt-0.11D/qa/rpc-tests/test_framework/test_framework.py", line 118, in main
    self.run_test()
  File "/home/error/bitcoinxt-0.11D/qa/rpc-tests/forknotify.py", line 41, in run_test
    self.nodes[0].generate(1)
  File "/home/error/bitcoinxt-0.11D/qa/rpc-tests/test_framework/authproxy.py", line 126, in __call__
    raise JSONRPCException(response['error'])
Stopping nodes
Failed
```

I think this is the same problem as in #101, only with a different test, but I would appreciate someone else (@dgenr8?) reviewing it.

Node debug logs are included:
[node0-regtest-debug.txt](https://github.com/bitcoinxt/bitcoinxt/files/52257/node0-regtest-debug.txt)
[node1-regtest-debug.txt](https://github.com/bitcoinxt/bitcoinxt/files/52256/node1-regtest-debug.txt)
[node2-regtest-debug.txt](https://github.com/bitcoinxt/bitcoinxt/files/52255/node2-regtest-debug.txt)

**Comment by dgenr8** [comment]
Looks like you're likely correct.  Can you apply the same change and check if it works?  If so, I'll add to the PR.

Out of curiousity, can you describe your system?

Any trouble with bigblocks.py, bip65-cltv-p2p.py, or bipdersig-p2p.py?  They also set block versions.

**Comment by error10** [comment]
Applying the same patch to forknotify.py results in a pass.

```
Running 2nd level testscript forknotify.py...
Initializing test directory /tmp/testcDSY6W
Stopping nodes
Cleaning up
Tests successful
```

This system, as noted before, is running Fedora 23 x86_64. The kernel is 4.2.6-301.fc23.x86_64 . It is a KVM (qemu 2.3.1) virtual machine with 4GB of RAM and 8 vCPUs, and the host system has a Xeon D 1540 processor with 64GB of RAM. 

I had no issues with bigblocks.py, bip65-cltv-p2p.py, or bipdersig-p2p.py. I did, however, also have a problem with pruning.py having a too-short timeout, which seems unrelated to this (and I was about to report it separately).

**Comment by error10** [comment]
Fixed in 0.11E.
